fvm/transactionStorageLimiter.go

```diff
@@ -0,0 +1,74 @@
+package fvm
+
+import (
+	"github.com/onflow/flow-go/fvm/state"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type TransactionStorageLimiter struct{}
+
+func NewTransactionStorageLimiter() *TransactionStorageLimiter {
+	return &TransactionStorageLimiter{}
+}
+
+func (d *TransactionStorageLimiter) Process(
+	_ *VirtualMachine,
+	_ Context,
+	_ *TransactionProcedure,
+	ledger state.Ledger,
+) error {
+	accounts := state.NewAccounts(ledger)
+
+	registerIds, _ := ledger.RegisterUpdates()
+
+	checked := map[string]struct{}{}
+
+	for _, id := range registerIds {
+		owner := id.Owner
+
+		if _, wasChecked := checked[owner]; wasChecked {
+			// we already checked this account, move on
+			continue
+		}
+		checked[owner] = struct{}{}
+
+		// is this an address?
+		// does it exist?
+		address, exists, err := isExistingAccount(accounts, owner)
```

Comment:
I'd split address parsing into a separate function rather than merging it with `isExistingAccount`

Reply:
👍